src/file.ts

```diff
@@ -2323,6 +2328,10 @@ class File extends ServiceObject<File> {
    *     future major-version release.
    * @param {string} [config.cname] The cname for this bucket, i.e.,
    *     "https://cdn.example.com".
+   * @param {string} [config.contentSha256] The SHA256 digest value for the request
+   *     content. If you provide this value, the client must provide the HTTP header
+   *     `X-Goog-Content-SHA256` with the same value in its request, and the service
+   *     will verify that the content's SHA256 digest matches that provided here.
```

Comment:
I'd prefer if there was no helper because this doesn't translate to V2 and can be confusing.

Reply:
So we'll leave the responsibility to the user's side to provide the `X-Goog-Content-SHA256` header and we will populate the signed payload section of the canonical request with the provided SHA256. Does that sound okay?

Reply:
+1, I understand this to be an advanced feature. The client should replace `UNSIGNED-PAYLOAD` with the value of header `X-Goog-Content-SHA256` when it's provided by the user.

Reply:
Thank you!